### DISCUSSION TITLE
command line arguments for opening to predefined timestamp.

### DISCUSSION THREAD

**Discussion by svefro** [mifi/lossless-cut#903] [Ideas]
Hi.
I use mpv video player with a lot of different scripts to skim trough my raw video really fast to see what i want to keep.

if i get to a clip i do not want to keep i press CTRL+SHIFT+D and the clip is gone.
When i get to a clip i want to cut i press CTRL+SHIFT-L to launch lossless cut with the file open.

Since losslesscut-win.exe takes some time to open i continue on to the next clip in mpv and maybe a few more than that.

When i have a few instances off losslessCut open in the background i switch from mpv to the opened instances off lossless cut.
i then know that every instance i have open of lossless cut contains one video clip i'm going to cut. but i basically have to watch trough the whole clip again to locate the first cut point.

when i'm done cutting i delete the source file from within losslessCut (if i'm sure i have what i want from the exported clips)

Since I'm using mpv it would be relatively easy to pass my **current timestamp** in mpv to losslessCut-win.exe but **is that possible?** if so what is the command line argument? i have tested "losslessCut-win.exe filename.mpg -t 01:02:34" for example but it does not work.

This would be really great for my workflow.

Also the 'Trash original source file' option resets every time i launch losslessCut. can you fix that so i **do not have to check this every time. Remembering to app restart is not enough since i close the instance when i'm done with the file.**

Hope you understand my workflow :)

If you have not tested mpv video player i highly recommend it due to it's scripting possibilities.
But it takes a while to set up. If you want i can send you a preconfigured version for testing.

**Comment by svefro** [comment]
i managed to work around this by creating a script that first creates a .llc file with the first cut position.

but i do not like that this has to overwrite any existing .llc file 

this however do not solve the issue with always having to click "delete source file"

and it does not move playback position to that timestamp in the video

```

mp.add_key_binding(nil, "open_llc_cut_here", function()
    
    local current_file_path = mp.get_property("path")
    
    function SplitFilename(strFilename)
        -- Returns the Path, Filename, and Extension as 3 values
        return string.match(strFilename, "(.-)([^\\]-([^\\%.]+))$")
    end

    local path,filename,extension = SplitFilename(current_file_path)
    local filename_withoutExtension = filename:gsub("."..extension,"")
    
    local string_in_file =  [[
        {
            version: 1,
            mediaFileName: ']]..filename_withoutExtension..[[',
            cutSegments: [
              {
                start: ]] .. mp.get_property("playback-time") .. [[,
                name: '',
              },
            ],
          }
        ]]

    -- filename = oldfilename minus extension + "-proj.llc"
    local file = io.open(path..filename_withoutExtension.."-proj.llc", "w")
    if file then
        file:write(string_in_file)

        --file:write(mp.get_property("time-pos"))
        --file:write(mp.get_property("playback-time"))
        --file:write(",,")
        file:close()
    end


    mp.commandv("run", "O:/OneDrive/Files/PortableApps/PortableApps/LosslessCut/LosslessCut-win.exe", current_file_path);
    mp.osd_message("Launching LosslessCut with first cut start at "..mp.get_property("time-pos"),4)

end
)
```

**Comment by mifi** [comment]
Hi. Im not sure how common your use case it, but losslesscut wasnt really designed for multiple concurrent indtances and reopening all the time. It’s actually lucky that it works. However I agree that a command line option like this could be useful and isnt very hard to implement.

i created a feature request here https://github.com/mifi/lossless-cut/issues/980

**Comment by svefro** [comment]
Great. it is a work flow that works for me. :)

Could you add a feature request for remembering "delete source file" option also.? (even if application closes and relaunches)

**Reply by mifi** [reply]
I added here #371
